### PR TITLE
[RBAC] `az ad sp credential reset`: fix weak credential generation

### PR DESCRIPTION
**Pull request by qianwens**
**Description**  

Password autogenerated by `az ad sp credential reset` is a Guid which is not secure.

**Testing Guide**  

az ad sp credential reset -n test
Expected result: password should be a string with at least one special character
**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S169

**Comment by jiasli** [comment]
Please title the PR with meaningful description, like "[Graph] az ad sp credential reset: fix weak credential generation", so that `HISTORY.rst` contains useful information for the user.

File: src/azure-cli/azure/cli/command_modules/role/custom.py [hunk 1]
```diff
@@ -1741,6 +1741,24 @@ def _gen_guid():
...
+    password += random.choice(string.digits)
+    password += random.choice(string.punctuation)
+
+    for i in range(length - 4):
+        password += random.choice(random_source)
+
+    password_list = list(password)
+    random.SystemRandom().shuffle(password_list)
+    password = ''.join(password_list)
+    return password
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/role/custom.py:1759]: If the code is borrowed from somewhere else, it is a good practice to put the source link in the comment: https://pynative.com/python-generate-random-string/

**Comment by jiasli** [comment]
@fengzhou-msft I think we have decided not to include backticks `` ` `` in the title?

File: src/azure-cli/azure/cli/command_modules/role/custom.py [hunk 2]
```diff
@@ -1741,6 +1741,25 @@ def _gen_guid():
...
+def _random_password(length):
+    import random
+    import string
+    random_source = string.ascii_letters + string.digits + string.punctuation
+    password = random.choice(string.ascii_lowercase)
+    password += random.choice(string.ascii_uppercase)
+    password += random.choice(string.digits)
+    password += random.choice(string.punctuation)
+
+    for i in range(length - 4):    # pylint: disable=unused-variable
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/role/custom.py:1754]:
```suggestion
    for _ in range(length - 4):
```

**Pull request merged**